### ISSUE TITLE
Update multi-chain contract API

### ISSUE THREAD

**Issue by volovyks** [near/mpc#473]
### Description
Propositions are welcomed, here is what I think we should change:
```[tasklist]
### TODO
- [x] Organize contract API, separate public API
- [x] return result as a struct (key values), not an array
- [ ] return derived user PK that was used in signing (can be done on FE, but it should allow to check the signature on the contract level and make FE life easier, we may have many integrations in the future)
- [ ] return root and user pk as a point or string, not near_sdk::PK
- [ ] ~~prevent payload hash rotation on the client side~~ (does not seem possible)
- [ ] https://github.com/near/mpc-recovery/issues/548
- [ ] https://github.com/near/mpc-recovery/issues/578
- [ ] https://github.com/near/mpc-recovery/issues/593
- [ ] https://github.com/near/mpc-recovery/issues/583
- [ ] https://github.com/near/mpc-recovery/issues/585
- [ ] https://github.com/near/mpc-recovery/issues/579
- [ ] Improve error handling, list all posible errors in enum or using [handle_result]
```

```[tasklist]
### Under discussion
- [ ] https://github.com/near/mpc-recovery/issues/513
```

**Comment by volovyks** [comment]
Update:
- switch from payload hash to payload and hashing function type.

**Comment by volovyks** [comment]
Blocked on payload vs payload hash decision
cc @DavidM-D

**Comment by volovyks** [comment]
We will continue sending only the hash of the payload.

**Comment by volovyks** [comment]
Closing in favour of #663
